aws/client/aws-client-core/src/main/java/software/amazon/smithy/java/aws/client/core/settings/S3EndpointSettings.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.aws.client.core.settings;
+
+import software.amazon.smithy.java.client.core.ClientSetting;
+import software.amazon.smithy.java.context.Context;
+
+/**
+ * Configures AWS specific endpoint settings.
+ */
+public interface S3EndpointSettings<B extends ClientSetting<B>> extends EndpointSettings<B> {
+    /**
+     * If the SDK client is configured to use S3 transfer acceleration, defaults to false.
+     */
+    Context.Key<Boolean> S3_ACCELERATE = Context.key("AWS::S3::Accelerate");
```

Comment:
The keys for `AccountId` and `Endpoint` are descriptive but these are the built-in names. Should be one or the other.